internal/daysteps/daysteps.go

```diff
@@ -1,13 +1,65 @@
 package daysteps
 
+import (
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/simenoff/go1fl-sprint5-final/internal/personaldata"
+	"github.com/simenoff/go1fl-sprint5-final/internal/spentenergy"
+)
+
 type DaySteps struct {
-	// TODO: добавить поля
+	Steps    int
+	Duration time.Duration
+	personaldata.Personal
 }
 
 func (ds *DaySteps) Parse(datastring string) (err error) {
-	// TODO: реализовать функцию
+
+	parts := strings.Split(datastring, ",")
+	if len(parts) != 2 {
+		return errors.New("неверный формат данных: ожидается два элемента")
+	}
+
+	//
+	steps, err := strconv.Atoi(parts[0])
+	if err != nil {
+		return fmt.Errorf("ошибка при парсинге количества шагов: %v", err)
+	}
+	if steps <= 0 {
+		return errors.New("количество шагов должно быть больше 0")
+	}
+
+	//
+	duration, err := time.ParseDuration(parts[1])
+	if err != nil {
+		return fmt.Errorf("ошибка при парсинге продолжительности: %v", err)
+	}
+	if duration <= 0 {
+		return errors.New("продолжительность должна быть больше 0")
+	}
+
+	ds.Steps = steps
+	ds.Duration = duration
+
+	return nil
 }
 
 func (ds DaySteps) ActionInfo() (string, error) {
-	// TODO: реализовать функцию
+
+	distanceKm := spentenergy.Distance(ds.Steps, ds.Height)
+
+	calories, err := spentenergy.WalkingSpentCalories(ds.Steps, ds.Weight, ds.Height, ds.Duration)
+	if err != nil {
+		return "", err
```

Comment:
Лучше всегда оборачивать
```suggestion
		return "", fmt.Errorf("error calculating calories: %w", err)
```